runtime/pattern.go

```diff
@@ -36,14 +36,16 @@ type Pattern struct {
 	tailLen int
 	// verb is the VERB part of the path pattern. It is empty if the pattern does not have VERB part.
 	verb string
+	// caseSensitive defines whether the pattern matching is caseSensitive or not
+	caseSensitive bool
 }
 
 // NewPattern returns a new Pattern from the given definition values.
 // "ops" is a sequence of op codes. "pool" is a constant pool.
 // "verb" is the verb part of the pattern. It is empty if the pattern does not have the part.
 // "version" must be 1 for now.
 // It returns an error if the given definition is invalid.
-func NewPattern(version int, ops []int, pool []string, verb string) (Pattern, error) {
+func NewPattern(version int, ops []int, pool []string, verb string, caseSensitive bool) (Pattern, error) {
```

Comment:
set the caseSensitive option in the Pattern struct, so we can change the Match behavior